pkg/alertmanager/alertmanager.go

```diff
@@ -367,3 +384,12 @@ func buildReceiverIntegrations(nc *config.Receiver, tmpl *template.Template, log
 	}
 	return integrations, nil
 }
+
+func md5HashAsMetricValue(data []byte) float64 {
+	sum := md5.Sum(data)
+	// We only want 48 bits as a float64 only has a 53 bit mantissa.
+	smallSum := sum[0:6]
+	var bytes = make([]byte, 8)
+	copy(bytes, smallSum)
+	return float64(binary.LittleEndian.Uint64(bytes))
+}
```

Comment:
This is super ugly. As Marco pointed out above, I'd suggest to expose hash as a label (with algorithm in the label name).